apps/web/src/components/Room/Room.tsx

```diff
@@ -33,8 +36,43 @@ const useStyles = makeStyles((theme: Theme) => {
 export default function Room() {
   const classes = useStyles();
   const { isChatWindowOpen } = useChatContext();
-  const { isBackgroundSelectionOpen } = useVideoContext();
-  const { appState } = useAppState();
+  const { isBackgroundSelectionOpen, room } = useVideoContext();
+  const { connect: playerConnect } = usePlayerContext();
+  const { registerUserDocument } = useSyncContext();
+  const { appState, appDispatch } = useAppState();
+  const speakerDepartedRef = useRef(false);
+
+  useEffect(() => {
+    if (room) {
+      const handleSpeakerDeparted = () => (speakerDepartedRef.current = true);
+      room.on('speakerDeparted', handleSpeakerDeparted);
+
+      return () => {
+        room.off('speakerDeparted', handleSpeakerDeparted);
+      };
+    }
+  }, [room]);
+
+  useEffect(() => {
+    if (room) {
+      speakerDepartedRef.current = false;
+      appDispatch({ type: 'set-is-loading', isLoading: true });
+
+      const handleConnectToPlayer = async () => {
+        if (!speakerDepartedRef.current) {
+          const { data } = await joinStreamAsViewer(room.localParticipant.identity, room.name);
+          await playerConnect(data.token);
+          await connectViewerToPlayer(appState.participantName, appState.eventName);
+          registerUserDocument(data.sync_object_names.user_document);
```

Comment:
May be good to display a snackbar to the user explaining that they were booted by the host.

Reply:
<img width="456" alt="Screen Shot 2022-01-21 at 2 36 45 PM" src="https://user-images.githubusercontent.com/77076398/150609275-51456bd8-d1d0-4fa2-9842-34128d65c4cd.png">